lib/resque/data_store.rb

```diff
@@ -80,7 +80,7 @@ def reconnect
     # is O(N) for the keyspace, so be careful - this can be slow for big databases.
     def all_resque_keys
       @redis.keys("*").map do |key|
-        key.sub("#{Resque.redis.namespace}:", '')
```

Comment:
This was only working since Resque.redis normally returns this instance and then namespace goes through method_missing.